### ISSUE TITLE
Whitespace on the left hand side of the  html `value` attribute adds a zero on the formatted value on page load

### ISSUE THREAD

**Issue by cotein** [autoNumeric/autoNumeric#479]
Hello people! My English is horrible, i sorry for that.

I have the follow problem.
Why the plugin put one zero at first place's number? 

Example: $ 0.456.661,20
Zero and dot afet dolar sign not shold be!

Thanks!!!

**Comment by AlexandreBonneau** [comment]
Hello @cotein, in order to help you, you would need to give us more information, mainly what is explained in the placeholder text when you created this issue:
- The version of AutoNumeric
- The option used
- The browser used
and, more importantly, a reproducable use case on codepen.io!

**Comment by estavex** [comment]
I also had that problem. I was using php to dynamically set the values. I figured out, it happens when you assign a string value into the `value` field, instead of integer value. Just use `value="intval($varname);"` if you're using php

**Comment by cotein** [comment]
Actually I do not know what happened I solved it by creating a custom
directive in Vue.js

Vue.directive ('currency', {
   Bind: function (el, binding) {
     //console.log(el, binding)
     Let anElement = new AutoNumeric (el, {{
DigitGroupSeparator: '.',
DigitalGroupSpacing: '3',
DecimalCharacter: ',',
CurrencySymbol: '$',
NegativeBracketsTypeOnBlur: '(,)',
MaximumValue: '999999999.99',
MinimumValue: '-999999999.99',
DecimalPlacesOverride: '2',
ModifyValueOnWheel: false,
ShowOnlyNumbersOnFocus: true,

Thanks!!!


2017-08-18 14:08 GMT-03:00 estavex <notifications@github.com>:

**Comment by AlexandreBonneau** [comment]
@cotein congrats!

I have a `vue-autoNumeric` components that works great, but I need to clean it up before open-sourcing it to Github, so stay tuned ;)

@estavex using the html attribute `value=` should normally not lead to any problem. If you could create a reproducible test case, I'll gladly check what could be the problem exactly. Thanks!

**Comment by estavex** [comment]
@AlexandreBonneau on page load, it writes a value as `$00.123,456,789` or something, but when i hover over it, it displays and stays as `$123,456,789`. I just put `intval($varname);` and only then it displayed as `$123,456,789` normally.

**Comment by AlexandreBonneau** [comment]
@estavex Hum, this is definitely not right, could you please open a [new issue](https://github.com/autoNumeric/autoNumeric/issues/new) with the setup you use so I can fix that? thanks

**Comment by estavex** [comment]
@AlexandreBonneau there's nothing special i use, i only followed the docs and setup basic autoNumeric.js functionality. this just happens when you set default values as the `value` attribute in html. I've tried to hardcode a number `value="123456789"` which is the integer value, and it displays normally as $123,456,789.00`. Then I tried `value="<?php echo $varname; ?>"` and it displays as `$0.123,456,789.00`. I then echo'ed printed 123456789 normally and echoed the $varname somewhere on the side. Both of those statements return 123456789 normally. Then I thought, I am echoing a string, so why not try to convert it to integer because integer seems to be working in `value` attribute. So I did, using php's function `echo intval($varname);` and Then it worked fine.

I think autoNumeric gets confused when it receives string value numbers, you probably built it to accept only integer-based numbers as input (which is obviously the right way). Nevertheless, it's a small bug and I think you shouldn't waste your time on fixing this

**Comment by AlexandreBonneau** [comment]
Well, it's weird because Html attributes do not know the difference between integer, float or strings ; they are [always interpreted as strings](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#Content_versus_IDL_attributes) : *" The content attribute is always a string even when the expected value should be an integer."*.

Have you tried to `var_dump($varname);` in order to be sure there is only an integer there?
Perhaps some non printable character got mixed in there? Perhaps an additional pair of double-quotes ?

**Comment by AlexandreBonneau** [comment]
👉 @estavex

**Comment by danad7583** [comment]
Hello,

I just had a similar problem when formatting some percentage display values. I had some Razor syntax inside of a span to help determine the values to display, which caused spaces to be added to the text value of the Html. AutoNumeric does a detection to see if the first digit of a value is numeric, and if it is not, it replaces it with a "0". After I trimmed up the spaces in the View between the Razor syntax and html elements, everything displayed correctly.

Hope this helps resolve the issue.

**Comment by AlexandreBonneau** [comment]
@danad7583 would you have a live example that reproduce the problem?
That would help a lot!

**Comment by AlexandreBonneau** [comment]
Ok, thanks for your comment @danad7583, it looks like indeed, having a space on the left-hand side of the Html `value` attribute results in an added zero.

Test cases:
```html
<input id="issue_479_left" type="text" placeholder="#479" value=" 42"> <!-- fail -->
<input id="issue_479_right" type="text" placeholder="#479" value="24 "> <!-- works -->
<input id="issue_479_both" type="text" placeholder="#479" value=" 22 "> <!-- fail -->
```

**Comment by AlexandreBonneau** [comment]
The fix is available in `v4.1.0-beta.8`!
